actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1727,8 +1727,8 @@ def id
       # <tt>aria-describedby</tt> attribute referencing the <tt><span></tt>
       # element, sharing a common <tt>id</tt> root (<tt>post_title</tt>, in this
       # case).
-      def field_id(method, *suffixes, index: @index)
-        @template.field_id(@object || @object_name, method, *suffixes, index: @index)
+      def field_id(method, *suffixes, index: nil)
+        @template.field_id(@object_name || @object, method, *suffixes, index: index || @index)
```

Comment:
The `index:` argument wasn't being used at all before, there's a much simpler fix for that:

```suggestion
      def field_id(method, *suffixes, index: @index)
        @template.field_id(@object_name || @object, method, *suffixes, index: index)
```

This obviously conflicts with the suggestion above, but hopefully you get the idea.